src/controllers/user.js

```diff
@@ -68,22 +68,21 @@ export const getAll = async (req, res) => {
 }
 
 export const updateById = async (req, res) => {
-  const paramsId = Number(req.params.id)
-  const { id } = req.user
-  const foundUser = await User.findById(paramsId)
+  const { firstName, lastName } = req.body
+  try {
+    validation.update(firstName, lastName)
+    const { id } = req.user
+    const canPatch = validation.validateCanModify(req)
+    if (!canPatch) {
+      return sendDataResponse(res, 403, { error: ERR.NOT_AUTHORISED })
+    }
+    const updatedUser = await User.updateUser(id, req.body)
+    delete updatedUser.password
```

Comment:
Instead of removing the password, create a new object that contains exactly what you want to return. It's much safer this way.

e.g.: think what would happen if the field changes name but you forget to update this line